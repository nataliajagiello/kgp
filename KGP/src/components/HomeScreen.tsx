import AsyncStorage from '@react-native-async-storage/async-storage';
import React, {useEffect, useState} from 'react';
import {ActivityIndicator, Text, View, StyleSheet} from 'react-native';
import {mountainsData} from '../data/mountains';
import Mountain from '../models/Mountain';
import MountainsSectionList from './MountainsSectionList';
import * as colors from '../styles/colors';

const styles = StyleSheet.create({
  activityIndicator: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    color: colors.main,
  },
  container: {
    flex: 1,
  },
  title: {
    fontSize: 30,
    textAlign: 'center',
    fontWeight: 'bold',
    padding: 10,
    backgroundColor: colors.main,
    color: 'white',
  },
});

const title = 'Korona GÃ³r Polski';

const HomeScreen = () => {
  const [mountains, setMountains] = useState<Mountain[]>();

  useEffect(() => {
    const isSummitConcquered = async (id: number) => {
      const checkBoxValue: string =
        (await AsyncStorage.getItem(`${id}`)) ?? 'false';
      return JSON.parse(checkBoxValue);
    };

    const completeData = async () => {
      for (const mountain of mountainsData) {
        mountain.concquered = await isSummitConcquered(mountain.id);
      }
      setMountains(mountainsData);
    };

    completeData();
  }, []);

  return (
    <View style={styles.container}>
      {mountains ? (
        <>
          <Text style={styles.title}>{title}</Text>
          <MountainsSectionList
            mountains={mountains}
            setMountains={setMountains}
          />
        </>
      ) : (
        <ActivityIndicator size="large" style={styles.activityIndicator} />
      )}
    </View>
  );
};

export default HomeScreen;
