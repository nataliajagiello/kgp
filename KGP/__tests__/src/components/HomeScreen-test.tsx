import 'react-native';
import React from 'react';
import HomeScreen from '../../../src/components/HomeScreen';

import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

describe('HomeScreen', () => {
  it('renders correctly', async () => {
    let tree;
    await renderer.act(async () => {
      tree = renderer.create(<HomeScreen />);
    });
    // @ts-ignore
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders app title', async () => {
    const {findByText} = render(<HomeScreen />);
    expect(await findByText('Korona GÃ³r Polski')).toBeTruthy();
  });

  //temporary disabled, becasue of calendar click
  // it('updates concquered qty on checkbox click', async () => {
  //   const {findByTestId, findByText} = render(<HomeScreen />);
  //   const checkbox = await findByTestId('m-7');
  //   await act(async () => {
  //     fireEvent(checkbox, 'onValueChange', true);
  //   });
  //   expect(await findByText('Zdobyte 1/28')).toBeTruthy();
  //   expect(await findByText('Do zdobycia 27/28')).toBeTruthy();
  // });
});
