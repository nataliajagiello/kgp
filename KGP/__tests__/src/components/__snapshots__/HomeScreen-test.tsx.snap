// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#2E4053",
        "color": "white",
        "fontSize": 30,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Korona Gór Polski
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [],
          "sectionTitle": "Zdobyte 0/28",
        },
        Object {
          "data": Array [
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 2499,
              "id": 1,
              "name": "Rysy",
              "range": "Tatry",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1725,
              "id": 2,
              "name": "Babia Góra",
              "range": "Beskid Żywiecki",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1603,
              "id": 3,
              "name": "Śnieżka",
              "range": "Karkonosze",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1423,
              "id": 4,
              "name": "Śnieżnik",
              "range": "Masyw Śnieżnika",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1346,
              "id": 5,
              "name": "Tarnica",
              "range": "Bieszczady Zachodnie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1310,
              "id": 6,
              "name": "Turbacz",
              "range": "Gorce",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1266,
              "id": 7,
              "name": "Radziejowa",
              "range": "Beskid Sądecki",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1257,
              "id": 8,
              "name": "Skrzyczne",
              "range": "Beskid Śląski",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1171,
              "id": 9,
              "name": "Mogielica",
              "range": "Beskid Wyspowy",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1126,
              "id": 10,
              "name": "Wysoka Kopa",
              "range": "Góry Izerskie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1106,
              "id": 11,
              "name": "Rudawiec",
              "range": "Góry Bialskie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1084,
              "id": 12,
              "name": "Orlica",
              "range": "Góry Orlickie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1050,
              "id": 13,
              "name": "Wysoka",
              "range": "Pieniny",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 1015,
              "id": 14,
              "name": "Wielka Sowa",
              "range": "Góry Sowie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 997,
              "id": 15,
              "name": "Lackowa",
              "range": "Beskid Niski",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 989,
              "id": 16,
              "name": "Kowadło",
              "range": "Góry Złote",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 977,
              "id": 17,
              "name": "Jagodna",
              "range": "Góry Bystrzyckie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 945,
              "id": 18,
              "name": "Skalnik",
              "range": "Rudawy Janowickie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 936,
              "id": 19,
              "name": "Waligóra",
              "range": "Góry Kamienne",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 933,
              "id": 20,
              "name": "Czupel",
              "range": "Beskid Mały",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 919,
              "id": 21,
              "name": "Szczeliniec Wielki",
              "range": "Góry Stołowe",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 904,
              "id": 22,
              "name": "Lubomir",
              "range": "Beskid Makowski",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 889,
              "id": 23,
              "name": "Biskupia Kopa",
              "range": "Góry Opawskie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 850,
              "id": 24,
              "name": "Chełmiec",
              "range": "Góry Wałbrzyskie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 765,
              "id": 25,
              "name": "Kłodzka Góra",
              "range": "Góry Bardzkie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 724,
              "id": 26,
              "name": "Skopiec",
              "range": "Góry Kaczawskie",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 718,
              "id": 27,
              "name": "Ślęża",
              "range": "Masyw Ślęży",
            },
            Object {
              "concquered": false,
              "date": undefined,
              "elevation": 614,
              "id": 28,
              "name": "Łysica",
              "range": "Góry Świętokrzyskie",
            },
          ],
          "sectionTitle": "Do zdobycia 28/28",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
        2,
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "backgroundColor": "#2E4053",
              "color": "white",
              "fontSize": 25,
              "fontWeight": "bold",
              "padding": 10,
              "textAlign": "center",
            }
          }
        >
          Zdobyte 0/28
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "backgroundColor": "#2E4053",
              "color": "white",
              "fontSize": 25,
              "fontWeight": "bold",
              "padding": 10,
              "textAlign": "center",
            }
          }
        >
          Do zdobycia 28/28
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                Object {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="m-1"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "marginLeft": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1.3,
                        },
                        Object {
                          "scaleY": 1.3,
                        },
                      ],
                    },
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Tatry
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Rysy
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              2499
               m n.p.m.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                Object {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="m-2"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "marginLeft": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1.3,
                        },
                        Object {
                          "scaleY": 1.3,
                        },
                      ],
                    },
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Beskid Żywiecki
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Babia Góra
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              1725
               m n.p.m.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                Object {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="m-3"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "marginLeft": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1.3,
                        },
                        Object {
                          "scaleY": 1.3,
                        },
                      ],
                    },
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Karkonosze
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Śnieżka
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              1603
               m n.p.m.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                Object {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="m-4"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "marginLeft": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1.3,
                        },
                        Object {
                          "scaleY": 1.3,
                        },
                      ],
                    },
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Masyw Śnieżnika
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Śnieżnik
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              1423
               m n.p.m.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                Object {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="m-5"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "marginLeft": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1.3,
                        },
                        Object {
                          "scaleY": 1.3,
                        },
                      ],
                    },
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Bieszczady Zachodnie
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Tarnica
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              1346
               m n.p.m.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                Object {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="m-6"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "marginLeft": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1.3,
                        },
                        Object {
                          "scaleY": 1.3,
                        },
                      ],
                    },
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Gorce
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Turbacz
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              1310
               m n.p.m.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                Object {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="m-7"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "marginLeft": 25,
                      "transform": Array [
                        Object {
                          "scaleX": 1.3,
                        },
                        Object {
                          "scaleY": 1.3,
                        },
                      ],
                    },
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Beskid Sądecki
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              Radziejowa
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "margin": 5,
                  "textAlign": "center",
                }
              }
            >
              1266
               m n.p.m.
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
